### DISCUSSION TITLE
Running plotter in headless environment

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#621] [Q&A] [answered]
Hello!!
Bit of a speculative question here. I am trying to run my code on a headless ubuntu box using the command line. As part of the flow of my program, it updates the the plotter and takes screenshot of the updated plotter using the `ve.plotter.Plotter.screenshot()` command. When it gets to this command, it fails with an "Aborted (core dumped)" error. 

I have come to realise that this is because at various points I am rendering to screen, which is causing issues for my headless box. I have utilised the "offscreen" keyword in my plotter.Plotter() call and in my plotter.show() call , so that the plotter window does not render (if I don't this leads to the same error). 

My question is this: Can I still take a screenshot without the screen being present? I know it sounds a bit odd but I was thinking that the data structure that represents the plotter window still exists, so perhaps that can be saved to an image somehow without rendering and tripping the core dump error. It is rather vital that I get those images, so not taking them isn't really an option...and you can save matplotlib plots to a file without rendering them on screen...am I thinking in the right terms? Please correct me if I have the wrong idea here, and, any suggestions would be greatly appreciated.

Thanks for all your help, 
j

**Comment by marcomusy** [comment, marked as answer]
Hi @jkissi have you tried already this:
https://github.com/marcomusy/vedo/issues/64#issuecomment-559979201

also:
https://github.com/marcomusy/vedo/issues/609#issuecomment-1057869173

**Reply by jkissi** [reply]
Goood news!! After a bit of wrangling, I believe (fingers crossed) this fixes my problem. My program appears to running through. this is fantastic. I have to add the bash code to my .bashrc file, as opposed to /etc/rc.local, for anyone who may happen on this thread. But currently looking good!! Once again, thanks marco, you're a legend!

Cheers, 
j
p.s. I am still working on that last question. I will get back to that thread, I havent forgotten about it! Cheers again!!
